python/phase-1.md

```diff
@@ -0,0 +1,380 @@
+* **NOTE: These will have proper docs and types when developed!**
+
+### Client
+
+Exported from `temporalio` (but probably in `temporalio/client/client.py`, exported from `temporalio.client` then
+exported from `temporalio`):
+
+```python
+class Client:
+    def __init__(
+        self,
+        addr: str,
+        *,
+        namespace: str = "default",
+        default_task_queue: str = None,
+        identity: str = None
+        # ... additional options omitted for brevity
+    ) -> None:
+        # Actually dials gRPC here
+        pass
+
+    # Raw gRPC service
+    @property
+    def service(self) -> workflowservice.WorkflowService:
+        pass
+
+    def start_workflow(
+        workflow: str,
+        *args,
+        id: str = None,
+        task_queue: str = None,
+        retry_policy: RetryPolicy = None,
+        # Note, we are embedding signal-with-start in here
+        start_signal: str = None,
+        start_signal_args=[],
+        # ... additional options omitted for brevity
+    ) -> WorkflowHandle[Any]:
+        pass
+
+    # In a future phase:
+    # @overload
+    # def start_workflow(workflow: Callable[P, T], *args: P.args, ...) -> WorkflowExecution[T]
+
+    # Does not validate whether a workflow exists
+    def workflow(
+        workflow_id: str,
+        run_id: str = None
+        # ...
+    ) -> WorkflowHandle[Any]:
+        pass
+
+
+class WorkflowHandle(Generic[T]):
+    def __init__(
+        self,
+        client: Client,
+        workflow_id: str,
+        *,
+        run_id: str = None
+        # ...
+    ) -> None:
+        pass
+
+    def __await__(self) -> T:
+        pass
+
+    async def cancel(self):
+        pass
+
+    async def describe(self):
+        pass
+
+    async def query(self, name: str, *args):
+        pass
+
+    async def signal(self, name: str, *args):
+        pass
+
+    async def terminate(self, *, reason: str = None):
+        pass
+```
+
+Notes:
+
+* We are intentionally defining a `Client` instead of a `WorkflowClient` since this will be general purpose
+  * Having a general purpose client keeps things simple, but at least for now I can't see defining any more than the two
+    methods and the one property
+  * We could make an abstract base class if we want for typing/purposes that just has the `service` property
+    unimplemented
+* We call it `start_workflow` instead of just `start` to be clearer
+* We have embedded `start_signal` and `start_signal_args` as parameters, so if the former is present we will perform a
+  signal with start. Is this ok?
+  * This is an intentional deviation from other SDKs, but makes logical sense
+* The client supports a default task queue, do we want this?
+
+### Worker
+
+Exported from `temporalio` (but probably in `temporalio/worker/worker.py`, exported from `temporalio.worker` then
+exported from `temporalio`):
+
+```python
+class Worker:
+    def __init__(
+        self,
+        client: temporalio.Client,
```

Comment:
You'll need a Core instance here, not a client.

Reply:
Derp, that is true! Not sure how I missed that.

Reply:
working on the go SDK for too long :)

Reply:
Updated